src/app/directives/resize-column.directive.ts

```diff
@@ -0,0 +1,144 @@
+import {ElementRef, Renderer2, HostListener, Directive, AfterViewInit, Input} from '@angular/core';
+import {SettingsService, SettingsProperties} from '../services/settings.service';
+import {Store} from 'src/app/stores/store';
+
+export type ColumnType = 'routeMenu' | 'envLogs';
+
+/**
+ * Allows the resizing of the parent element using css(width,)
+ * On mousedown registers listeners into the body (mousemove,mouseup) to detect resizing.
+ *
+ * Min width is passed as a parameter and max is calculated based on body size and a factor.
+ */
+
+@Directive({
+  // tslint:disable-next-line:directive-selector
+  selector: 'resize-column'
+})
+export class ResizeColumnDirective implements AfterViewInit {
+
+  constructor(private elementRef: ElementRef,
+              private settingsService: SettingsService,
+              private store: Store,
+              private renderer: Renderer2) {
+    elementRef.nativeElement.style.cursor = 'col-resize';
+  }
+
+  // Max factor to calculate max width based on body width
+  private static readonly MAX_WIDTH_FACTOR = 0.8;
+
+  @Input() type: ColumnType;
+  @Input() minWidth = 100; // min width in pixels
+  @Input() minWidthFromRight: number; // max width that should remain
+
+  // Event removers for mousemove / mouseup events to body
+  private mouseMoveRemover: Function;
+  private mouseUpRemover: Function;
+
+  private pressed: boolean;
+  // The x point where the mousedown event occurred
+  private startX;
+  private startWidth;
+
+  ngAfterViewInit() {
+    // Init and set size from settings - if that exists
+    const settings = this.store.get('settings');
+    if (typeof settings !== 'undefined') {
+      let width;
+      if (this.type === 'routeMenu') {
+        width = settings.routeMenuSize;
+      } else {
+        width = settings.envLogsSize;
+      }
+
+      if (typeof width !== 'undefined') { // finally update width if needed
+        this.applyWidthCss(width);
+      }
+    }
```

Comment:
We could eventually have a race condition where settings are not yet loaded. 
Please use something like this instead:

```suggestion
    this.store.select('settings').pipe(
      filter(settings => !!settings),
      first(),
    ).subscribe((settings) => {
      let width;
      if (this.type === 'routeMenu') {
        width = settings.routeMenuSize;
      } else {
        width = settings.envLogsSize;
      }

      if (typeof width !== 'undefined') { // finally update width if needed
        this.applyWidthCss(width);
      }
    });
```